diff -up krita-4.1.7.101/cmake/modules/FindExiv2.cmake.1~ krita-4.1.7.101/cmake/modules/FindExiv2.cmake
--- krita-4.1.7.101/cmake/modules/FindExiv2.cmake.1~	2018-12-16 13:44:52.000000000 +0100
+++ krita-4.1.7.101/cmake/modules/FindExiv2.cmake	2018-12-25 19:52:47.319249424 +0100
@@ -57,7 +57,7 @@ find_library(EXIV2_LIBRARY NAMES exiv2 l
 
 # Get the version number from exiv2/version.hpp and store it in the cache:
 if(EXIV2_INCLUDE_DIR  AND NOT  EXIV2_VERSION)
-  file(READ ${EXIV2_INCLUDE_DIR}/exiv2/version.hpp EXIV2_VERSION_CONTENT)
+  file(READ ${EXIV2_INCLUDE_DIR}/exiv2/exv_conf.h EXIV2_VERSION_CONTENT)
   string(REGEX MATCH "#define EXIV2_MAJOR_VERSION +\\( *([0-9]+) *\\)"  _dummy "${EXIV2_VERSION_CONTENT}")
   set(EXIV2_VERSION_MAJOR "${CMAKE_MATCH_1}")
 
diff -up krita-4.1.7.101/libs/ui/kisexiv2/kis_xmp_io.cpp.1~ krita-4.1.7.101/libs/ui/kisexiv2/kis_xmp_io.cpp
--- krita-4.1.7.101/libs/ui/kisexiv2/kis_xmp_io.cpp.1~	2018-12-16 13:44:52.000000000 +0100
+++ krita-4.1.7.101/libs/ui/kisexiv2/kis_xmp_io.cpp	2018-12-25 20:07:22.383581095 +0100
@@ -17,7 +17,7 @@
 #include "kis_xmp_io.h"
 
 #include <string>
-#include <exiv2/xmp.hpp>
+#include <exiv2/xmp_exiv2.hpp>
 
 #include "kis_exiv2.h"
 
@@ -277,9 +277,8 @@ bool KisXMPIO::loadFrom(KisMetaData::Sto
                 const Exiv2::XmpArrayValue* xav = dynamic_cast<const Exiv2::XmpArrayValue*>(value.get());
                 Q_ASSERT(xav);
                 QList<KisMetaData::Value> array;
-                for (std::vector< std::string >::const_iterator it = xav->value_.begin();
-                        it != xav->value_.end(); ++it) {
-                    QString value = it->c_str();
+		for (int it = 0; it < xav->count(); it++) {
+                    QString value = QString::fromStdString(xav->toString(it));
                     if (parser) {
                         array.push_back(parser->parse(value));
                     } else {
