diff -up krita-5.0.8/3rdparty_plugins/ext_gmic/CMakeLists.txt.5~ krita-5.0.8/3rdparty_plugins/ext_gmic/CMakeLists.txt
--- krita-5.0.8/3rdparty_plugins/ext_gmic/CMakeLists.txt.5~	2022-05-26 11:24:07.203787118 +0200
+++ krita-5.0.8/3rdparty_plugins/ext_gmic/CMakeLists.txt	2022-05-26 11:24:58.900804982 +0200
@@ -9,7 +9,7 @@ ExternalProject_Add( ext_gmic
 
     INSTALL_DIR ${PREFIX_ext_gmic}
 
-    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${PREFIX_ext_gmic} -DGMIC_QT_HOST=krita-plugin -DENABLE_SYSTEM_GMIC=FALSE -DCMAKE_BUILD_TYPE=${GLOBAL_BUILD_TYPE} ${GLOBAL_PROFILE}
+    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${PREFIX_ext_gmic} -DGMIC_QT_HOST=krita-plugin -DENABLE_SYSTEM_GMIC=FALSE -DCMAKE_BUILD_TYPE=${GLOBAL_BUILD_TYPE} ${GLOBAL_PROFILE} -DKIS_IMAGE_INTERFACE_DIR=${INSTALL_ROOT}/include/
     LIST_SEPARATOR "|"
 
     UPDATE_COMMAND ""
