diff -up krita-5.0.0/CMakeLists.txt.1~ krita-5.0.0/CMakeLists.txt
--- krita-5.0.0/CMakeLists.txt.1~	2021-12-20 11:16:15.000000000 +0100
+++ krita-5.0.0/CMakeLists.txt	2021-12-23 13:55:17.310666833 +0100
@@ -993,16 +993,17 @@ set_package_properties(Poppler PROPERTIE
 ##
 ## Test for quazip
 ##
-find_package(QuaZip 0.6)
+find_package(QuaZip-Qt5 REQUIRED)
 set_package_properties(QuaZip PROPERTIES
     DESCRIPTION "A library for reading and writing zip files"
     URL "https://stachenov.github.io/quazip/"
     TYPE REQUIRED
     PURPOSE "Needed for reading and writing KRA and ORA files"
 )
+include_directories(/usr/include/QuaZip-Qt5-1.1/quazip)
 
 # FIXME: better way to do this?
-list (APPEND ANDROID_EXTRA_LIBS ${QUAZIP_LIBRARIES}
+list (APPEND ANDROID_EXTRA_LIBS QuaZip::QuaZip
                                 ${EXPAT_LIBRARY}
                                 ${KF5_LIBRARIES}/libssl_1_1.so
                                 ${KF5_LIBRARIES}/libcrypto_1_1.so
diff -up krita-5.0.0/libs/store/CMakeLists.txt.1~ krita-5.0.0/libs/store/CMakeLists.txt
--- krita-5.0.0/libs/store/CMakeLists.txt.1~	2021-12-20 11:16:15.000000000 +0100
+++ krita-5.0.0/libs/store/CMakeLists.txt	2021-12-23 13:52:47.546275503 +0100
@@ -1,3 +1,4 @@
+find_package(QuaZip-Qt5 REQUIRED)
 include_directories(${QUAZIP_INCLUDE_DIRS})
 
 add_subdirectory(tests)
@@ -24,7 +25,7 @@ target_link_libraries(kritastore
         KF5::ConfigCore
         Qt5::Xml 
         Qt5::Gui 
-        ${QUAZIP_LIBRARIES}
+        QuaZip::QuaZip
 )
 
 set_target_properties(kritastore PROPERTIES
diff -up krita-5.0.0/CMakeLists.txt.omv~ krita-5.0.0/CMakeLists.txt
--- krita-5.0.0/CMakeLists.txt.omv~	2021-12-23 15:02:24.337737295 +0100
+++ krita-5.0.0/CMakeLists.txt	2021-12-23 15:03:20.118777693 +0100
@@ -1001,6 +1001,7 @@ set_package_properties(QuaZip PROPERTIES
     PURPOSE "Needed for reading and writing KRA and ORA files"
 )
 include_directories(/usr/include/QuaZip-Qt5-1.1/quazip)
+set(QUAZIP_LIBRARIES -lquazip1-qt5)
 
 # FIXME: better way to do this?
 list (APPEND ANDROID_EXTRA_LIBS QuaZip::QuaZip
diff -up krita-5.0.0/libs/resources/CMakeLists.txt.omv~ krita-5.0.0/libs/resources/CMakeLists.txt
