diff -Naur a/sdk/tests/simpletest.h b/sdk/tests/simpletest.h
--- a/sdk/tests/simpletest.h
+++ b/sdk/tests/simpletest.h
@@ -19,7 +19,6 @@
     app.setAttribute(Qt::AA_Use96Dpi, true); \
     /*QLocale en_US(QLocale::English, QLocale::UnitedStates); \
     KLocalizedString::setLanguages(QStringList() << QStringLiteral("en_US"));*/ \
-    QTEST_DISABLE_KEYPAD_NAVIGATION \
     TestObject tc; \
     QTEST_SET_MAIN_SOURCE_PATH \
     return QTest::qExec(&tc, argc, argv);
