diff -up krita-4.2.9/benchmarks/kis_stroke_benchmark.cpp.omv~ krita-4.2.9/benchmarks/kis_stroke_benchmark.cpp
--- krita-4.2.9/benchmarks/kis_stroke_benchmark.cpp.omv~	2020-04-08 02:18:39.209330237 +0200
+++ krita-4.2.9/benchmarks/kis_stroke_benchmark.cpp	2020-04-08 02:18:45.874429899 +0200
@@ -27,6 +27,7 @@ inline double drand48()
 #endif
 
 #include <QTest>
+#include <QPainterPath>
 
 #include "kis_stroke_benchmark.h"
 #include "kis_benchmark_values.h"
diff -up krita-4.2.9/libs/basicflakes/tools/KoCreatePathTool_p.h.omv~ krita-4.2.9/libs/basicflakes/tools/KoCreatePathTool_p.h
--- krita-4.2.9/libs/basicflakes/tools/KoCreatePathTool_p.h.omv~	2020-04-08 01:55:23.994216781 +0200
+++ krita-4.2.9/libs/basicflakes/tools/KoCreatePathTool_p.h	2020-04-08 01:55:33.759345331 +0200
@@ -33,6 +33,8 @@
 #include <KoViewConverter.h>
 #include "kis_config.h"
 
+#include <QPainterPath>
+
 #include "math.h"
 
 class KoStrokeConfigWidget;
diff -up krita-4.2.9/libs/brush/kis_auto_brush.cpp.omv~ krita-4.2.9/libs/brush/kis_auto_brush.cpp
--- krita-4.2.9/libs/brush/kis_auto_brush.cpp.omv~	2020-04-08 01:41:13.543619215 +0200
+++ krita-4.2.9/libs/brush/kis_auto_brush.cpp	2020-04-08 01:41:24.667674759 +0200
@@ -31,6 +31,7 @@
 #include <QBuffer>
 #include <QFile>
 #include <QFileInfo>
+#include <QPainterPath>
 
 #include <KoColor.h>
 #include <KoColorSpace.h>
diff -up krita-4.2.9/libs/brush/kis_boundary.cc.omv~ krita-4.2.9/libs/brush/kis_boundary.cc
--- krita-4.2.9/libs/brush/kis_boundary.cc.omv~	2020-04-08 01:40:48.999465407 +0200
+++ krita-4.2.9/libs/brush/kis_boundary.cc	2020-04-08 01:40:53.138523778 +0200
@@ -18,6 +18,7 @@
 
 #include "kis_boundary.h"
 #include <QPainter>
+#include <QPainterPath>
 #include <QPen>
 
 #include "KoColorSpace.h"
diff -up krita-4.2.9/libs/brush/kis_brush.cpp.omv~ krita-4.2.9/libs/brush/kis_brush.cpp
--- krita-4.2.9/libs/brush/kis_brush.cpp.omv~	2020-04-08 01:40:26.501062459 +0200
+++ krita-4.2.9/libs/brush/kis_brush.cpp	2020-04-08 01:40:34.010197112 +0200
@@ -28,6 +28,7 @@
 #include <QPoint>
 #include <QFileInfo>
 #include <QBuffer>
+#include <QPainterPath>
 
 #include <kis_debug.h>
 #include <klocalizedstring.h>
diff -up krita-4.2.9/libs/flake/KoClipMask.cpp.omv~ krita-4.2.9/libs/flake/KoClipMask.cpp
--- krita-4.2.9/libs/flake/KoClipMask.cpp.omv~	2020-04-08 01:33:17.162984555 +0200
+++ krita-4.2.9/libs/flake/KoClipMask.cpp	2020-04-08 01:33:21.789076844 +0200
@@ -21,6 +21,7 @@
 #include <QRectF>
 #include <QTransform>
 #include <QPainter>
+#include <QPainterPath>
 #include <KoShape.h>
 #include "kis_algebra_2d.h"
 
diff -up krita-4.2.9/libs/flake/KoClipMaskPainter.cpp.omv~ krita-4.2.9/libs/flake/KoClipMaskPainter.cpp
--- krita-4.2.9/libs/flake/KoClipMaskPainter.cpp.omv~	2020-04-08 01:35:26.589514946 +0200
+++ krita-4.2.9/libs/flake/KoClipMaskPainter.cpp	2020-04-08 01:35:29.689574347 +0200
@@ -19,6 +19,7 @@
 #include "KoClipMaskPainter.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QRectF>
 
 #include "kis_assert.h"
diff -up krita-4.2.9/libs/flake/KoConnectionShape.cpp.omv~ krita-4.2.9/libs/flake/KoConnectionShape.cpp
--- krita-4.2.9/libs/flake/KoConnectionShape.cpp.omv~	2020-04-08 01:28:24.385774891 +0200
+++ krita-4.2.9/libs/flake/KoConnectionShape.cpp	2020-04-08 01:28:26.591824971 +0200
@@ -33,7 +33,7 @@
 #include <KoXmlWriter.h>
 #include <KoXmlNS.h>
 #include <KoUnit.h>
-#include <QPainter>
+#include <QPainterPath>
 
 #include <FlakeDebug.h>
 
diff -up krita-4.2.9/libs/flake/KoGradientBackground.cpp.omv~ krita-4.2.9/libs/flake/KoGradientBackground.cpp
--- krita-4.2.9/libs/flake/KoGradientBackground.cpp.omv~	2020-04-08 01:35:01.912040215 +0200
+++ krita-4.2.9/libs/flake/KoGradientBackground.cpp	2020-04-08 01:35:05.432108139 +0200
@@ -31,6 +31,7 @@
 #include <QSharedPointer>
 #include <QBrush>
 #include <QPainter>
+#include <QPainterPath>
 
 class KoGradientBackgroundPrivate : public KoShapeBackgroundPrivate
 {
diff -up krita-4.2.9/libs/flake/KoHatchBackground.cpp.omv~ krita-4.2.9/libs/flake/KoHatchBackground.cpp
--- krita-4.2.9/libs/flake/KoHatchBackground.cpp.omv~	2020-04-08 01:32:44.474328119 +0200
+++ krita-4.2.9/libs/flake/KoHatchBackground.cpp	2020-04-08 01:32:47.578390785 +0200
@@ -36,6 +36,7 @@
 #include <QColor>
 #include <QString>
 #include <QPainter>
+#include <QPainterPath>
 
 class KoHatchBackgroundPrivate : public KoColorBackgroundPrivate
 {
diff -up krita-4.2.9/libs/flake/KoOdfGradientBackground.cpp.omv~ krita-4.2.9/libs/flake/KoOdfGradientBackground.cpp
--- krita-4.2.9/libs/flake/KoOdfGradientBackground.cpp.omv~	2020-04-08 01:27:44.157850883 +0200
+++ krita-4.2.9/libs/flake/KoOdfGradientBackground.cpp	2020-04-08 01:32:14.497719186 +0200
@@ -32,6 +32,7 @@
 #include <KoOdfStylesReader.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QColor>
 #include <QImage>
 #include <qmath.h>
diff -up krita-4.2.9/libs/flake/KoPathShape.cpp.omv~ krita-4.2.9/libs/flake/KoPathShape.cpp
--- krita-4.2.9/libs/flake/KoPathShape.cpp.omv~	2020-04-08 01:31:45.376120854 +0200
+++ krita-4.2.9/libs/flake/KoPathShape.cpp	2020-04-08 01:34:48.223775410 +0200
@@ -50,6 +50,7 @@
 
 #include <FlakeDebug.h>
 #include <QPainter>
+#include <QPainterPath>
 
 #include "kis_global.h"
 
diff -up krita-4.2.9/libs/flake/KoPatternBackground.cpp.omv~ krita-4.2.9/libs/flake/KoPatternBackground.cpp
--- krita-4.2.9/libs/flake/KoPatternBackground.cpp.omv~	2020-04-08 01:32:30.239039804 +0200
+++ krita-4.2.9/libs/flake/KoPatternBackground.cpp	2020-04-08 01:32:34.099118134 +0200
@@ -37,6 +37,7 @@
 
 #include <QBrush>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPointer>
 
 class KoPatternBackgroundPrivate : public KoShapeBackgroundPrivate
diff -up krita-4.2.9/libs/flake/KoShapeManager.cpp.omv~ krita-4.2.9/libs/flake/KoShapeManager.cpp
--- krita-4.2.9/libs/flake/KoShapeManager.cpp.omv~	2020-04-08 01:26:58.125766956 +0200
+++ krita-4.2.9/libs/flake/KoShapeManager.cpp	2020-04-08 01:31:33.037865251 +0200
@@ -49,6 +49,7 @@
 #include <QApplication>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QTimer>
 #include <FlakeDebug.h>
 
diff -up krita-4.2.9/libs/flake/KoShapeShadow.cpp.omv~ krita-4.2.9/libs/flake/KoShapeShadow.cpp
--- krita-4.2.9/libs/flake/KoShapeShadow.cpp.omv~	2020-04-08 01:27:16.454202094 +0200
+++ krita-4.2.9/libs/flake/KoShapeShadow.cpp	2020-04-08 01:33:51.108658376 +0200
@@ -32,6 +32,7 @@
 #include <KoViewConverter.h>
 #include <FlakeDebug.h>
 #include <QPainter>
+#include <QPainterPath>
 #include <QAtomicInt>
 #include <QImage>
 #include <QRectF>
diff -up krita-4.2.9/libs/flake/KoSnapGuide.cpp.omv~ krita-4.2.9/libs/flake/KoSnapGuide.cpp
--- krita-4.2.9/libs/flake/KoSnapGuide.cpp.omv~	2020-04-08 01:33:03.005701196 +0200
+++ krita-4.2.9/libs/flake/KoSnapGuide.cpp	2020-04-08 01:33:07.687795066 +0200
@@ -27,6 +27,7 @@
 #include <KoCanvasBase.h>
 
 #include <QPainter>
+#include <QPainterPath>
 
 #include <math.h>
 
diff -up krita-4.2.9/libs/flake/KoSnapStrategy.cpp.omv~ krita-4.2.9/libs/flake/KoSnapStrategy.cpp
--- krita-4.2.9/libs/flake/KoSnapStrategy.cpp.omv~	2020-04-08 01:33:31.812276300 +0200
+++ krita-4.2.9/libs/flake/KoSnapStrategy.cpp	2020-04-08 01:33:34.982339238 +0200
@@ -27,6 +27,7 @@
 #include <KoViewConverter.h>
 
 #include <QPainter>
+#include <QPainterPath>
 
 #include <cmath>
 
diff -up krita-4.2.9/libs/flake/svg/SvgParser.cpp.omv~ krita-4.2.9/libs/flake/svg/SvgParser.cpp
--- krita-4.2.9/libs/flake/svg/SvgParser.cpp.omv~	2020-04-08 01:29:43.241529623 +0200
+++ krita-4.2.9/libs/flake/svg/SvgParser.cpp	2020-04-08 01:29:45.658582316 +0200
@@ -30,7 +30,7 @@
 #include <FlakeDebug.h>
 
 #include <QColor>
-#include <QPainter>
+#include <QPainterPath>
 #include <QDir>
 
 #include <KoShape.h>
diff -up krita-4.2.9/libs/flake/tests/TestKoMarkerCollection.cpp.omv~ krita-4.2.9/libs/flake/tests/TestKoMarkerCollection.cpp
--- krita-4.2.9/libs/flake/tests/TestKoMarkerCollection.cpp.omv~	2020-04-08 01:38:10.561592793 +0200
+++ krita-4.2.9/libs/flake/tests/TestKoMarkerCollection.cpp	2020-04-08 01:38:14.519665639 +0200
@@ -21,6 +21,7 @@
 #include <QTest>
 #include <QFileInfo>
 #include <QPainter>
+#include <QPainterPath>
 #include <KoMarker.h>
 #include <KoMarkerCollection.h>
 #include <KoPathShape.h>
diff -up krita-4.2.9/libs/flake/tests/TestPointMergeCommand.cpp.omv~ krita-4.2.9/libs/flake/tests/TestPointMergeCommand.cpp
--- krita-4.2.9/libs/flake/tests/TestPointMergeCommand.cpp.omv~	2020-04-08 01:39:49.697399980 +0200
+++ krita-4.2.9/libs/flake/tests/TestPointMergeCommand.cpp	2020-04-08 01:39:57.269536629 +0200
@@ -24,6 +24,7 @@
 #include "KoPathPointData.h"
 #include <sdk/tests/kistest.h>
 #include <QTest>
+#include <QPainterPath>
 #include <FlakeDebug.h>
 
 void TestPointMergeCommand::closeSingleLinePath()
diff -up krita-4.2.9/libs/flake/tests/TestShapeBackgroundCommand.cpp.omv~ krita-4.2.9/libs/flake/tests/TestShapeBackgroundCommand.cpp
--- krita-4.2.9/libs/flake/tests/TestShapeBackgroundCommand.cpp.omv~	2020-04-08 01:38:32.864002465 +0200
+++ krita-4.2.9/libs/flake/tests/TestShapeBackgroundCommand.cpp	2020-04-08 01:38:40.465141654 +0200
@@ -23,6 +23,7 @@
 #include "KoColorBackground.h"
 #include "KoShapePaintingContext.h"
 #include "KoViewConverter.h"
+#include <QPainterPath>
 
 #include <QTest>
 
diff -up krita-4.2.9/libs/flake/tests/TestSnapStrategy.cpp.omv~ krita-4.2.9/libs/flake/tests/TestSnapStrategy.cpp
--- krita-4.2.9/libs/flake/tests/TestSnapStrategy.cpp.omv~	2020-04-08 01:39:04.729584570 +0200
+++ krita-4.2.9/libs/flake/tests/TestSnapStrategy.cpp	2020-04-08 01:39:13.771749087 +0200
@@ -19,6 +19,7 @@
 
 #include "TestSnapStrategy.h"
 #include <QTest>
+#include <QPainterPath>
 #include "KoSnapStrategy.h"
 #include "KoPathShape.h"
 #include "KoSnapProxy.h"
diff -up krita-4.2.9/libs/flake/tests/TestSvgParser.cpp.omv~ krita-4.2.9/libs/flake/tests/TestSvgParser.cpp
--- krita-4.2.9/libs/flake/tests/TestSvgParser.cpp.omv~	2020-04-08 01:36:12.191383666 +0200
+++ krita-4.2.9/libs/flake/tests/TestSvgParser.cpp	2020-04-08 01:36:24.082608468 +0200
@@ -22,7 +22,7 @@
 #include <QTest>
 #include <svg/SvgUtil.h>
 #include <KoShapeStrokeModel.h>
-
+#include <QPainterPath>
 
 #include "SvgParserTestingUtils.h"
 
diff -up krita-4.2.9/libs/flake/tests/TestSvgText.cpp.omv~ krita-4.2.9/libs/flake/tests/TestSvgText.cpp
--- krita-4.2.9/libs/flake/tests/TestSvgText.cpp.omv~	2020-04-08 01:37:00.289288811 +0200
+++ krita-4.2.9/libs/flake/tests/TestSvgText.cpp	2020-04-08 01:37:09.268456605 +0200
@@ -28,6 +28,7 @@
 #include <SvgLoadingContext.h>
 #include <SvgGraphicContext.h>
 #include <QFont>
+#include <QPainterPath>
 
 void addProp(SvgLoadingContext &context,
              KoSvgTextProperties &props,
diff -up krita-4.2.9/libs/flake/text/KoSvgTextChunkShape_p.h.omv~ krita-4.2.9/libs/flake/text/KoSvgTextChunkShape_p.h
--- krita-4.2.9/libs/flake/text/KoSvgTextChunkShape_p.h.omv~	2020-04-08 01:31:04.373266352 +0200
+++ krita-4.2.9/libs/flake/text/KoSvgTextChunkShape_p.h	2020-04-08 01:31:11.039406276 +0200
@@ -22,6 +22,7 @@
 #include "KoSvgTextProperties.h"
 #include <KoShapeContainer_p.h>
 #include <QTextCharFormat>
+#include <QPainterPath>
 
 class SvgGraphicsContext;
 
diff -up krita-4.2.9/libs/flake/text/KoSvgTextShape.cpp.omv~ krita-4.2.9/libs/flake/text/KoSvgTextShape.cpp
--- krita-4.2.9/libs/flake/text/KoSvgTextShape.cpp.omv~	2020-04-08 01:30:11.326138152 +0200
+++ krita-4.2.9/libs/flake/text/KoSvgTextShape.cpp	2020-04-08 01:30:30.958558823 +0200
@@ -47,7 +47,8 @@
 #include <QThread>
 #include <vector>
 #include <memory>
-#include <QPainter>
+#include <QPainterPath>
+#include <QFontMetrics>
 #include <boost/optional.hpp>
 
 #include <text/KoSvgTextChunkShapeLayoutInterface.h>
diff -up krita-4.2.9/libs/flake/tools/KoPathTool.cpp.omv~ krita-4.2.9/libs/flake/tools/KoPathTool.cpp
--- krita-4.2.9/libs/flake/tools/KoPathTool.cpp.omv~	2020-04-08 01:29:26.272157924 +0200
+++ krita-4.2.9/libs/flake/tools/KoPathTool.cpp	2020-04-08 01:29:28.563208288 +0200
@@ -65,7 +65,7 @@
 #include <QAction>
 #include <FlakeDebug.h>
 #include <klocalizedstring.h>
-#include <QPainter>
+#include <QPainterPath>
 #include <QBitmap>
 #include <QTabWidget>
 
diff -up krita-4.2.9/libs/global/KisHandlePainterHelper.cpp.omv~ krita-4.2.9/libs/global/KisHandlePainterHelper.cpp
--- krita-4.2.9/libs/global/KisHandlePainterHelper.cpp.omv~	2020-04-08 01:25:27.174531524 +0200
+++ krita-4.2.9/libs/global/KisHandlePainterHelper.cpp	2020-04-08 01:25:31.434639259 +0200
@@ -18,7 +18,7 @@
 
 #include "KisHandlePainterHelper.h"
 
-#include <QPainter>
+#include <QPainterPath>
 #include "kis_algebra_2d.h"
 #include "kis_painting_tweaks.h"
 
diff -up krita-4.2.9/libs/image/brushengine/kis_paintop_settings.cpp.omv~ krita-4.2.9/libs/image/brushengine/kis_paintop_settings.cpp
--- krita-4.2.9/libs/image/brushengine/kis_paintop_settings.cpp.omv~	2020-04-08 01:37:38.507000552 +0200
+++ krita-4.2.9/libs/image/brushengine/kis_paintop_settings.cpp	2020-04-08 01:37:49.469203558 +0200
@@ -22,6 +22,7 @@
 #include <QImage>
 #include <QColor>
 #include <QPointer>
+#include <QPainterPath>
 
 #include <KoPointerEvent.h>
 #include <KoColor.h>
diff -up krita-4.2.9/libs/ui/flake/kis_shape_controller.cpp.omv~ krita-4.2.9/libs/ui/flake/kis_shape_controller.cpp
diff -up krita-4.2.9/libs/ui/flake/kis_shape_selection.h.omv~ krita-4.2.9/libs/ui/flake/kis_shape_selection.h
--- krita-4.2.9/libs/ui/flake/kis_shape_selection.h.omv~	2020-04-08 01:46:52.341201866 +0200
+++ krita-4.2.9/libs/ui/flake/kis_shape_selection.h	2020-04-08 01:47:01.254289808 +0200
@@ -18,6 +18,8 @@
 #ifndef KIS_SHAPE_SELECTION_H
 #define KIS_SHAPE_SELECTION_H
 
+#include <QPainterPath>
+
 #include <KoShapeLayer.h>
 #include <KoShapeFactoryBase.h>
 #include <KoShapeUserData.h>
diff -up krita-4.2.9/libs/ui/kis_painting_assistants_decoration.cpp.omv~ krita-4.2.9/libs/ui/kis_painting_assistants_decoration.cpp
--- krita-4.2.9/libs/ui/kis_painting_assistants_decoration.cpp.omv~	2020-04-08 01:49:01.384548073 +0200
+++ krita-4.2.9/libs/ui/kis_painting_assistants_decoration.cpp	2020-04-08 01:49:12.994676246 +0200
@@ -33,6 +33,7 @@
 #include "KisViewManager.h"
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QApplication>
 
 struct KisPaintingAssistantsDecoration::Private {
diff -up krita-4.2.9/libs/ui/kis_selection_decoration.h.omv~ krita-4.2.9/libs/ui/kis_selection_decoration.h
--- krita-4.2.9/libs/ui/kis_selection_decoration.h.omv~	2020-04-08 01:49:51.003103082 +0200
+++ krita-4.2.9/libs/ui/kis_selection_decoration.h	2020-04-08 01:50:04.904261852 +0200
@@ -22,6 +22,7 @@
 #include <QTimer>
 #include <QPolygon>
 #include <QPen>
+#include <QPainterPath>
 
 #include <kis_signal_compressor.h>
 #include "canvas/kis_canvas_decoration.h"
diff -up krita-4.2.9/libs/ui/tool/kis_shape_tool_helper.cpp.omv~ krita-4.2.9/libs/ui/tool/kis_shape_tool_helper.cpp
--- krita-4.2.9/libs/ui/tool/kis_shape_tool_helper.cpp.omv~	2020-04-08 01:50:31.791572774 +0200
+++ krita-4.2.9/libs/ui/tool/kis_shape_tool_helper.cpp	2020-04-08 01:50:50.391790709 +0200
@@ -23,6 +23,7 @@
 #include <KoShapeFactoryBase.h>
 #include <KoProperties.h>
 
+#include <QPainterPath>
 
 KoShape* KisShapeToolHelper::createRectangleShape(const QRectF& rect, qreal roundCornersX, qreal roundCornersY)
 {
diff -up krita-4.2.9/libs/ui/tool/kis_tool_paint.h.omv~ krita-4.2.9/libs/ui/tool/kis_tool_paint.h
--- krita-4.2.9/libs/ui/tool/kis_tool_paint.h.omv~	2020-04-08 01:51:18.208120773 +0200
+++ krita-4.2.9/libs/ui/tool/kis_tool_paint.h	2020-04-08 01:51:32.842296340 +0200
@@ -24,6 +24,7 @@
 #include <QGridLayout>
 #include <QTimer>
 #include <QCheckBox>
+#include <QPainterPath>
 
 #include <KoCanvasResourceProvider.h>
 #include <KoToolBase.h>
diff -up krita-4.2.9/libs/ui/widgets/kis_cie_tongue_widget.cpp.omv~ krita-4.2.9/libs/ui/widgets/kis_cie_tongue_widget.cpp
--- krita-4.2.9/libs/ui/widgets/kis_cie_tongue_widget.cpp.omv~	2020-04-08 01:52:11.425765219 +0200
+++ krita-4.2.9/libs/ui/widgets/kis_cie_tongue_widget.cpp	2020-04-08 01:52:20.904881692 +0200
@@ -40,6 +40,7 @@ wavelength, and thus define the outline
 
 #include <QPointF>
 #include <QPainter>
+#include <QPainterPath>
 #include <QFile>
 #include <QTimer>
 #include <QPaintEvent>
diff -up krita-4.2.9/libs/ui/widgets/kis_curve_widget.cpp.omv~ krita-4.2.9/libs/ui/widgets/kis_curve_widget.cpp
--- krita-4.2.9/libs/ui/widgets/kis_curve_widget.cpp.omv~	2020-04-08 01:53:34.337799778 +0200
+++ krita-4.2.9/libs/ui/widgets/kis_curve_widget.cpp	2020-04-08 01:53:43.956921991 +0200
@@ -27,6 +27,7 @@
 
 #include <QPixmap>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPoint>
 #include <QPen>
 #include <QEvent>
diff -up krita-4.2.9/libs/ui/widgets/kis_tone_curve_widget.cpp.omv~ krita-4.2.9/libs/ui/widgets/kis_tone_curve_widget.cpp
--- krita-4.2.9/libs/ui/widgets/kis_tone_curve_widget.cpp.omv~	2020-04-08 01:52:42.434148018 +0200
+++ krita-4.2.9/libs/ui/widgets/kis_tone_curve_widget.cpp	2020-04-08 01:52:54.428297440 +0200
@@ -25,6 +25,7 @@
 #include <QPointF>
 #include <QPolygonF>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPaintEvent>
 #include <QImage>
 #include <cmath>
diff -up krita-4.2.9/plugins/assistants/Assistants/ConcentricEllipseAssistant.cc.omv~ krita-4.2.9/plugins/assistants/Assistants/ConcentricEllipseAssistant.cc
--- krita-4.2.9/plugins/assistants/Assistants/ConcentricEllipseAssistant.cc.omv~	2020-04-08 01:56:45.723303107 +0200
+++ krita-4.2.9/plugins/assistants/Assistants/ConcentricEllipseAssistant.cc	2020-04-08 01:56:50.046361193 +0200
@@ -23,6 +23,7 @@
 #include <klocalizedstring.h>
 #include "kis_debug.h"
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 #include <kis_canvas2.h>
diff -up krita-4.2.9/plugins/assistants/Assistants/EllipseAssistant.cc.omv~ krita-4.2.9/plugins/assistants/Assistants/EllipseAssistant.cc
--- krita-4.2.9/plugins/assistants/Assistants/EllipseAssistant.cc.omv~	2020-04-08 01:58:03.349354908 +0200
+++ krita-4.2.9/plugins/assistants/Assistants/EllipseAssistant.cc	2020-04-08 01:58:12.342477888 +0200
@@ -23,6 +23,7 @@
 #include <klocalizedstring.h>
 #include "kis_debug.h"
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -up krita-4.2.9/plugins/assistants/Assistants/FisheyePointAssistant.cc.omv~ krita-4.2.9/plugins/assistants/Assistants/FisheyePointAssistant.cc
--- krita-4.2.9/plugins/assistants/Assistants/FisheyePointAssistant.cc.omv~	2020-04-08 01:57:33.017941794 +0200
+++ krita-4.2.9/plugins/assistants/Assistants/FisheyePointAssistant.cc	2020-04-08 01:57:44.408096620 +0200
@@ -25,6 +25,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -up krita-4.2.9/plugins/assistants/Assistants/InfiniteRulerAssistant.cc.omv~ krita-4.2.9/plugins/assistants/Assistants/InfiniteRulerAssistant.cc
--- krita-4.2.9/plugins/assistants/Assistants/InfiniteRulerAssistant.cc.omv~	2020-04-08 01:58:30.013720178 +0200
+++ krita-4.2.9/plugins/assistants/Assistants/InfiniteRulerAssistant.cc	2020-04-08 01:58:38.971843315 +0200
@@ -25,6 +25,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -up krita-4.2.9/plugins/assistants/Assistants/kis_assistant_tool.cc.omv~ krita-4.2.9/plugins/assistants/Assistants/kis_assistant_tool.cc
--- krita-4.2.9/plugins/assistants/Assistants/kis_assistant_tool.cc.omv~	2020-04-08 01:56:18.390937240 +0200
+++ krita-4.2.9/plugins/assistants/Assistants/kis_assistant_tool.cc	2020-04-08 01:56:22.116986969 +0200
@@ -21,6 +21,7 @@
 #include <kis_assistant_tool.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QXmlStreamReader>
 #include <QXmlStreamWriter>
 #include <QStandardPaths>
diff -up krita-4.2.9/plugins/assistants/Assistants/ParallelRulerAssistant.cc.omv~ krita-4.2.9/plugins/assistants/Assistants/ParallelRulerAssistant.cc
--- krita-4.2.9/plugins/assistants/Assistants/ParallelRulerAssistant.cc.omv~	2020-04-08 01:59:36.442638099 +0200
+++ krita-4.2.9/plugins/assistants/Assistants/ParallelRulerAssistant.cc	2020-04-08 01:59:45.863769138 +0200
@@ -25,6 +25,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -up krita-4.2.9/plugins/assistants/Assistants/PerspectiveAssistant.cc.omv~ krita-4.2.9/plugins/assistants/Assistants/PerspectiveAssistant.cc
--- krita-4.2.9/plugins/assistants/Assistants/PerspectiveAssistant.cc.omv~	2020-04-08 01:59:09.857269439 +0200
+++ krita-4.2.9/plugins/assistants/Assistants/PerspectiveAssistant.cc	2020-04-08 01:59:18.342386919 +0200
@@ -24,6 +24,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -up krita-4.2.9/plugins/assistants/Assistants/RulerAssistant.cc.omv~ krita-4.2.9/plugins/assistants/Assistants/RulerAssistant.cc
--- krita-4.2.9/plugins/assistants/Assistants/RulerAssistant.cc.omv~	2020-04-08 02:00:07.719073892 +0200
+++ krita-4.2.9/plugins/assistants/Assistants/RulerAssistant.cc	2020-04-08 02:00:17.190206284 +0200
@@ -23,6 +23,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -up krita-4.2.9/plugins/assistants/Assistants/SplineAssistant.cc.omv~ krita-4.2.9/plugins/assistants/Assistants/SplineAssistant.cc
--- krita-4.2.9/plugins/assistants/Assistants/SplineAssistant.cc.omv~	2020-04-08 02:01:18.859072840 +0200
+++ krita-4.2.9/plugins/assistants/Assistants/SplineAssistant.cc	2020-04-08 02:01:27.232191071 +0200
@@ -23,6 +23,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -up krita-4.2.9/plugins/assistants/Assistants/VanishingPointAssistant.cc.omv~ krita-4.2.9/plugins/assistants/Assistants/VanishingPointAssistant.cc
--- krita-4.2.9/plugins/assistants/Assistants/VanishingPointAssistant.cc.omv~	2020-04-08 02:00:40.484532709 +0200
+++ krita-4.2.9/plugins/assistants/Assistants/VanishingPointAssistant.cc	2020-04-08 02:00:48.893650821 +0200
@@ -25,6 +25,7 @@
 #include <klocalizedstring.h>
 
 #include <QPainter>
+#include <QPainterPath>
 #include <QLinearGradient>
 #include <QTransform>
 
diff -up krita-4.2.9/plugins/dockers/histogram/histogramdockerwidget.cpp.omv~ krita-4.2.9/plugins/dockers/histogram/histogramdockerwidget.cpp
--- krita-4.2.9/plugins/dockers/histogram/histogramdockerwidget.cpp.omv~	2020-04-08 02:04:06.469462079 +0200
+++ krita-4.2.9/plugins/dockers/histogram/histogramdockerwidget.cpp	2020-04-08 02:04:14.758581345 +0200
@@ -24,6 +24,7 @@
 #include <algorithm>
 #include <QTime>
 #include <QPainter>
+#include <QPainterPath>
 #include <functional>
 
 #include "KoChannelInfo.h"
diff -up krita-4.2.9/plugins/flake/pathshapes/enhancedpath/EnhancedPathShape.cpp.omv~ krita-4.2.9/plugins/flake/pathshapes/enhancedpath/EnhancedPathShape.cpp
--- krita-4.2.9/plugins/flake/pathshapes/enhancedpath/EnhancedPathShape.cpp.omv~	2020-04-08 02:04:48.631069639 +0200
+++ krita-4.2.9/plugins/flake/pathshapes/enhancedpath/EnhancedPathShape.cpp	2020-04-08 02:05:09.959377844 +0200
@@ -38,6 +38,8 @@
 #include <KoOdfWorkaround.h>
 #include <KoPathPoint.h>
 
+#include <QPainterPath>
+
 EnhancedPathShape::EnhancedPathShape(const QRect &viewBox)
     : m_viewBox(viewBox)
     , m_viewBoxOffset(0.0, 0.0)
diff -up krita-4.2.9/plugins/flake/textshape/TextShape.cpp.omv~ krita-4.2.9/plugins/flake/textshape/TextShape.cpp
--- krita-4.2.9/plugins/flake/textshape/TextShape.cpp.omv~	2020-04-08 02:04:32.637838901 +0200
+++ krita-4.2.9/plugins/flake/textshape/TextShape.cpp	2020-04-08 02:04:42.523981493 +0200
@@ -57,6 +57,7 @@
 #include <QApplication>
 #include <QFont>
 #include <QPainter>
+#include <QPainterPath>
 #include <QPen>
 #include <QTextLayout>
 
diff -up krita-4.2.9/plugins/paintops/curvebrush/curve_brush.cpp.omv~ krita-4.2.9/plugins/paintops/curvebrush/curve_brush.cpp
--- krita-4.2.9/plugins/paintops/curvebrush/curve_brush.cpp.omv~	2020-04-08 02:08:51.137602463 +0200
+++ krita-4.2.9/plugins/paintops/curvebrush/curve_brush.cpp	2020-04-08 02:08:58.168705696 +0200
@@ -21,6 +21,8 @@
 #include <KoColor.h>
 #include <KoColorSpace.h>
 
+#include <QPainterPath>
+
 #include <kis_random_accessor_ng.h>
 
 #include <cmath>
diff -up krita-4.2.9/plugins/paintops/curvebrush/kis_curve_paintop.cpp.omv~ krita-4.2.9/plugins/paintops/curvebrush/kis_curve_paintop.cpp
--- krita-4.2.9/plugins/paintops/curvebrush/kis_curve_paintop.cpp.omv~	2020-04-08 02:11:19.952794856 +0200
+++ krita-4.2.9/plugins/paintops/curvebrush/kis_curve_paintop.cpp	2020-04-08 02:11:27.694909303 +0200
@@ -21,6 +21,7 @@
 #include <cmath>
 
 #include <QRect>
+#include <QPainterPath>
 
 #include <kis_image.h>
 #include <kis_debug.h>
diff -up krita-4.2.9/plugins/paintops/experiment/kis_experiment_paintop.h.omv~ krita-4.2.9/plugins/paintops/experiment/kis_experiment_paintop.h
--- krita-4.2.9/plugins/paintops/experiment/kis_experiment_paintop.h.omv~	2020-04-08 02:14:43.944820568 +0200
+++ krita-4.2.9/plugins/paintops/experiment/kis_experiment_paintop.h	2020-04-08 02:14:52.647950070 +0200
@@ -19,6 +19,8 @@
 #ifndef KIS_EXPERIMENT_PAINTOP_H_
 #define KIS_EXPERIMENT_PAINTOP_H_
 
+#include <QPainterPath>
+
 #include <klocalizedstring.h>
 #include <brushengine/kis_paintop.h>
 #include <kis_types.h>
diff -up krita-4.2.9/plugins/tools/defaulttool/defaulttool/DefaultTool.cpp.omv~ krita-4.2.9/plugins/tools/defaulttool/defaulttool/DefaultTool.cpp
--- krita-4.2.9/plugins/tools/defaulttool/defaulttool/DefaultTool.cpp.omv~	2020-04-08 02:16:41.175567249 +0200
+++ krita-4.2.9/plugins/tools/defaulttool/defaulttool/DefaultTool.cpp	2020-04-08 02:16:52.897742154 +0200
@@ -71,6 +71,7 @@
 #include <QAction>
 #include <QKeyEvent>
 #include <QSignalMapper>
+#include <QPainterPath>
 #include <KoResourcePaths.h>
 
 #include <KoCanvasController.h>
diff -up krita-4.2.9/plugins/tools/defaulttool/defaulttool/SelectionDecorator.cpp.omv~ krita-4.2.9/plugins/tools/defaulttool/defaulttool/SelectionDecorator.cpp
--- krita-4.2.9/plugins/tools/defaulttool/defaulttool/SelectionDecorator.cpp.omv~	2020-04-08 02:18:01.812771254 +0200
+++ krita-4.2.9/plugins/tools/defaulttool/defaulttool/SelectionDecorator.cpp	2020-04-08 02:18:10.728904495 +0200
@@ -34,6 +34,8 @@
 #include <KoCanvasBase.h>
 #include <KoSvgTextShape.h>
 
+#include <QPainterPath>
+
 #include "kis_painting_tweaks.h"
 #include "kis_coordinates_converter.h"
 #include "kis_icon_utils.h"
diff -up krita-4.2.9/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp.omv~ krita-4.2.9/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp
--- krita-4.2.9/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp.omv~	2020-04-08 02:21:10.407593536 +0200
+++ krita-4.2.9/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphicShape.cpp	2020-04-08 02:21:17.796704263 +0200
@@ -28,6 +28,7 @@
 
 #include <QDebug>
 #include <QColor>
+#include <QPainterPath>
 
 #include <cmath>
 #include <cstdlib>
diff -up krita-4.2.9/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h.omv~ krita-4.2.9/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h
--- krita-4.2.9/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h.omv~	2020-04-08 02:20:15.188766402 +0200
+++ krita-4.2.9/plugins/tools/karbonplugins/tools/CalligraphyTool/KarbonCalligraphyTool.h	2020-04-08 02:20:21.003853478 +0200
@@ -23,6 +23,7 @@
 #include <KoToolBase.h>
 #include <KoPathShape.h>
 #include <QPointer>
+#include <QPainterPath>
 
 #include "KarbonCalligraphyOptionWidget.h"
 
diff -up krita-4.2.9/plugins/tools/svgtexttool/SvgTextTool.cpp.omv~ krita-4.2.9/plugins/tools/svgtexttool/SvgTextTool.cpp
--- krita-4.2.9/plugins/tools/svgtexttool/SvgTextTool.cpp.omv~	2020-04-08 02:18:51.912520196 +0200
+++ krita-4.2.9/plugins/tools/svgtexttool/SvgTextTool.cpp	2020-04-08 02:18:59.822638501 +0200
@@ -31,6 +31,7 @@
 #include <QGroupBox>
 #include <QFontDatabase>
 #include <QButtonGroup>
+#include <QPainterPath>
 
 #include <klocalizedstring.h>
 
diff -up krita-4.2.9/plugins/tools/tool_transform2/kis_liquify_paint_helper.cpp.omv~ krita-4.2.9/plugins/tools/tool_transform2/kis_liquify_paint_helper.cpp
--- krita-4.2.9/plugins/tools/tool_transform2/kis_liquify_paint_helper.cpp.omv~	2020-04-08 02:19:26.668040115 +0200
+++ krita-4.2.9/plugins/tools/tool_transform2/kis_liquify_paint_helper.cpp	2020-04-08 02:19:32.884133132 +0200
@@ -28,6 +28,8 @@
 #include "kis_liquify_paintop.h"
 #include "kis_liquify_properties.h"
 
+#include <QPainterPath>
+
 struct KisLiquifyPaintHelper::Private
 {
     Private(const KisCoordinatesConverter *_converter)
diff -up krita-4.2.9/plugins/tools/tool_transform2/kis_liquify_transform_strategy.cpp.omv~ krita-4.2.9/plugins/tools/tool_transform2/kis_liquify_transform_strategy.cpp
--- krita-4.2.9/plugins/tools/tool_transform2/kis_liquify_transform_strategy.cpp.omv~	2020-04-08 02:14:25.832551155 +0200
+++ krita-4.2.9/plugins/tools/tool_transform2/kis_liquify_transform_strategy.cpp	2020-04-08 02:14:30.015613364 +0200
@@ -22,6 +22,7 @@
 
 #include <QPointF>
 #include <QPainter>
+#include <QPainterPath>
 
 #include "KoPointerEvent.h"
 
diff -up krita-4.2.9/plugins/tools/tool_transform2/kis_perspective_transform_strategy.cpp.omv~ krita-4.2.9/plugins/tools/tool_transform2/kis_perspective_transform_strategy.cpp
--- krita-4.2.9/plugins/tools/tool_transform2/kis_perspective_transform_strategy.cpp.omv~	2020-04-08 02:15:41.940684080 +0200
+++ krita-4.2.9/plugins/tools/tool_transform2/kis_perspective_transform_strategy.cpp	2020-04-08 02:15:46.706755098 +0200
@@ -20,6 +20,7 @@
 
 #include <QPointF>
 #include <QPainter>
+#include <QPainterPath>
 #include <QMatrix4x4>
 #include <QVector2D>
 
diff -up krita-4.2.9/plugins/tools/tool_transform2/kis_transform_strategy_base.cpp.omv~ krita-4.2.9/plugins/tools/tool_transform2/kis_transform_strategy_base.cpp
--- krita-4.2.9/plugins/tools/tool_transform2/kis_transform_strategy_base.cpp.omv~	2020-04-08 02:18:21.831070431 +0200
+++ krita-4.2.9/plugins/tools/tool_transform2/kis_transform_strategy_base.cpp	2020-04-08 02:18:30.269196574 +0200
@@ -20,6 +20,7 @@
 
 #include <QImage>
 #include <QTransform>
+#include <QPainterPath>
 #include "KoPointerEvent.h"
 
 
diff -up krita-4.2.9/plugins/tools/tool_transform2/kis_transform_utils.h.omv~ krita-4.2.9/plugins/tools/tool_transform2/kis_transform_utils.h
--- krita-4.2.9/plugins/tools/tool_transform2/kis_transform_utils.h.omv~	2020-04-08 02:17:30.156298364 +0200
+++ krita-4.2.9/plugins/tools/tool_transform2/kis_transform_utils.h	2020-04-08 02:17:39.120432243 +0200
@@ -25,6 +25,7 @@
 
 #include <QTransform>
 #include <QMatrix4x4>
+#include <QPainterPath>
 #include <kis_processing_visitor.h>
 #include <limits>
 
