diff -up krita-5.2.0/libs/image/CMakeLists.txt.1~ krita-5.2.0/libs/image/CMakeLists.txt
--- krita-5.2.0/libs/image/CMakeLists.txt.1~	2023-10-05 13:58:33.707289895 +0200
+++ krita-5.2.0/libs/image/CMakeLists.txt	2023-10-05 14:02:26.614879921 +0200
@@ -379,6 +379,8 @@ target_link_libraries(kritaimage
 
         Eigen3::Eigen
         Boost::boost
+
+	atomic
 )
 
 if(HAVE_CXX_ATOMICS_WITH_LIB OR HAVE_CXX_ATOMICS64_WITH_LIB)
