diff -uraN krita-5.0.0-beta1/CMakeLists.txt omv-krita-5.0.0-beta1/CMakeLists.txt
--- krita-5.0.0-beta1/CMakeLists.txt	2021-08-18 04:17:52.000000000 +0200
+++ omv-krita-5.0.0-beta1/CMakeLists.txt	2021-08-22 15:33:50.627730012 +0200
@@ -993,16 +993,10 @@
 ##
 ## Test for quazip
 ##
-find_package(QuaZip 0.6)
-set_package_properties(QuaZip PROPERTIES
-    DESCRIPTION "A library for reading and writing zip files"
-    URL "https://stachenov.github.io/quazip/"
-    TYPE REQUIRED
-    PURPOSE "Needed for reading and writing KRA and ORA files"
-)
+find_package(QuaZip-Qt5 REQUIRED)
 
 # FIXME: better way to do this?
-list (APPEND ANDROID_EXTRA_LIBS ${QUAZIP_LIBRARIES}
+list (APPEND ANDROID_EXTRA_LIBS QuaZip::QuaZip
                                 ${EXPAT_LIBRARY}
                                 ${KF5_LIBRARIES}/libssl_1_1.so
                                 ${KF5_LIBRARIES}/libcrypto_1_1.so
diff -uraN krita-5.0.0-beta1/libs/store/CMakeLists.txt omv-krita-5.0.0-beta1/libs/store/CMakeLists.txt
--- krita-5.0.0-beta1/libs/store/CMakeLists.txt	2021-08-18 04:17:52.000000000 +0200
+++ omv-krita-5.0.0-beta1/libs/store/CMakeLists.txt	2021-08-22 15:32:41.741572244 +0200
@@ -1,3 +1,4 @@
+find_package(QuaZip-Qt5 REQUIRED)
 include_directories(${QUAZIP_INCLUDE_DIRS})
 
 add_subdirectory(tests)
@@ -24,7 +25,7 @@
         KF5::ConfigCore
         Qt5::Xml 
         Qt5::Gui 
-        ${QUAZIP_LIBRARIES}
+       QuaZip::QuaZip
 )
 
 set_target_properties(kritastore PROPERTIES
